### ISSUE TITLE
Types with duplicated names across multiple namespaces do not map

### ISSUE THREAD

**Issue by andrew-boyarshin** [SharpGenTools/SharpGenTools#181]
Feels like we have a type deduplication malfunctioning. Some rogue HashSet comparator? Vortice.DXGI.WinUI for reference.

```xml
  <include file="windows.ui.xaml.media.dxinterop.h" namespace="Vortice.DXGI" attach="true" />
  <include file="microsoft.ui.xaml.media.dxinterop.h" namespace="Vortice.DXGI.WinUI" output="WinUI" />
```

Does not generate `IVirtualSurfaceImageSourceNative` and the rest that are duplicated across these headers. TODO: Investigate for 2.0.

**Comment by andrew-boyarshin** [comment]
@amerkoleci FYI this is a blocker for Vortice 2.0 migration.

**Comment by amerkoleci** [comment]
Right, when I added WinUI bindings i had to manually wrap those types as the namespace was ignored and only the name was considered

**Comment by andrew-boyarshin** [comment]
Yeah. After a brief investigation, that's actually headers fault. :) `__IVirtualSurfaceImageSourceNative_FWD_DEFINED__` and the rest of the preprocessor definitions are the same in both `windows.*.h` and `microsoft.*.h`, so including them both from the same header will yield only the definition from the first header in the inclusion order. There is no way to fix that, unless we provide a way to split generated `.h` file (`CppHeaderGenerator`) into multiple, process each with CastXML separately (`CastXmlRunner`), then merge the results (`CppParser`). That's non-trivial, and I have no time to implement this in the foreseeable future.

**Comment by amerkoleci** [comment]
I started new Vortice.WinUI but types doesn't get generated even in new assembly, my Vortice.WinUI depends on Vortice.DXGI which blocks the WinUI types generations cause __ISwapChainBackgroundPanelNative_INTERFACE_DEFINED__ etc.

How can I workaround this?

**Comment by amerkoleci** [comment]
I've tried to workaround by modifying the header with:
__ISwapChainBackgroundPanelNative_WINUI_FWD_DEFINED__
__ISwapChainBackgroundPanelNative_WINUI_INTERFACE_DEFINED__

Adding the _WINUI_ but it doesn't work as well

**Comment by amerkoleci** [comment]
I've had to add manually all code with just GUID changed: https://github.com/amerkoleci/Vortice.Windows/commit/1b769b68387caa2dc6d9bb11664f7a55526b8055
